Further updates for MSFS support

mjh65 opened the pull request:
Two of these commits are refactoring updates to simplify a future update to support SQL-backed NAV data, and extend the runway and comms modelling to include data from the MSFS simulator world/scenery library.

The map overlay commit adds some additional capabilities which will be made use of by the SQL-backed NAV data, and also by a separate update to improve the overlayed map performance.

fpw commented on src/world/World.h:
```diff
@@ -37,10 +37,20 @@ class World {
 
-    virtual void visitNodes(const world::Location &upLeft, const world::Location &lowRight, NodeAcceptor f) = 0;
+    virtual int countNodes(const world::Location &upLeft, const world::Location &lowRight) = 0;
+    virtual void visitNodes(const world::Location &upLeft, const world::Location &lowRight, NodeAcceptor calllback, NodeVisitFilter filter) = 0;
```
Is the new filter used at any calling site yet? Because I don't think this kind of flag usage is legal in C++ - combining two enum values with an operator will result in an int, and in C++ there is no implicit cast back to an enum. Minimal example here: https://godbolt.org/z/75Ph58nK1

fpw replied:
Also, all enum values are the same: 0b1 == 0b0001 :-)

mjh65 replied:
No, the filter isn't used yet, only the combined value is passed in. And thanks for spotting the enum bitmask mistake. If I'd tested the filter I might have seen those issues first!
I'll change the enum to a set of const values, and fix the value definitions too.

mjh65 commented:
PR has been updated following review.
I have removed the enum used for filter selection and replaced it with int constants (which are now correctly encoded!)
I have also tested this code using individual and combinations of flags.
